### ISSUE TITLE
"Type 'keyof TProps' is not assignable to type 'string'" error with TypeScript  2.9

### ISSUE THREAD

**Issue by ulrichb** [facebook/immutable-js#1540]
After updating to TS 2.9 I got the following compile error:

```
.../node_modules/immutable/dist/immutable-nonambient.d.ts
(2381,30): A type predicate's type must be assignable to its parameter's type.
  Type 'keyof TProps' is not assignable to type 'string'.
    Type 'string | number | symbol' is not assignable to type 'string'.
      Type 'number' is not assignable to type 'string'.
```
The offset corresponds [to this line](https://github.com/facebook/immutable-js/blob/e65e5af806ea23a32ccf8f56c6fabf39605bac80/type-definitions/Immutable.d.ts#L2443).

Workaround: Use the `--keyofStringsOnly` compiler flag.

See also [keyof now includes string, number and symbol keys](https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#keyof-now-includes-string-number-and-symbol-keys)

TS version: 2.9.1, immutable: 4.0 RC 9

**Comment by bobby-stripe** [comment]
There is a PR that addresses this at #1539, still waiting on feedback.

**Comment by leebyron** [comment]
Fixed by #1593
